python-sdk/tests/test_sdk.py

```diff
@@ -0,0 +1,210 @@
+        run_language_model_check=False,
+        run_vector_check=True,
+        max_heuristic_score=0.5,
+        max_model_score=0.8,
+        max_vector_score=0.0,
+        injection_detected=False,
+    )
+    assert hasattr(rebuff_response, "heuristic_score")
+    assert hasattr(rebuff_response, "openai_score")
+    assert hasattr(rebuff_response, "vector_score")
+    assert hasattr(rebuff_response, "run_heuristic_check")
+    assert hasattr(rebuff_response, "run_language_model_check")
+    assert hasattr(rebuff_response, "run_vector_check")
+    assert hasattr(rebuff_response, "max_heuristic_score")
+    assert hasattr(rebuff_response, "max_model_score")
+    assert hasattr(rebuff_response, "max_vector_score")
+    assert hasattr(rebuff_response, "injection_detected")
+
+
+def test_add_canary_word(rebuff: RebuffSdk, user_inputs: List[str]):
+    for user_input in user_inputs:
+        prompt = f"Tell me a joke about\n{user_input}"
+        buffed_prompt, canary_word = rebuff.add_canary_word(prompt)
+        assert canary_word in buffed_prompt
+
+
+@pytest.mark.parametrize(
+    "canary_word_leaked",
+    [True, False],
+    ids=["canary_word_leaked", "canary_word_not_leaked"],
+)
+def test_is_canary_word_leaked(
+    rebuff: RebuffSdk, user_inputs: List[str], canary_word_leaked: bool
+):
+    for user_input in user_inputs:
+        prompt = f"Tell me a joke about\n{user_input}"
+        _, canary_word = rebuff.add_canary_word(prompt)
+        log_outcome = False
+        if canary_word_leaked:
+            response_completion = (
+                f"<!-- {canary_word} -->\nTell me a joke about\n{user_input}"
+            )
+            leak_detected = rebuff.is_canary_word_leaked(
+                user_input, response_completion, canary_word, log_outcome
+            )
+            assert leak_detected
+
+        else:
+            response_completion = f"Tell me a joke about\n{user_inputs}"
+
+            leak_detected = rebuff.is_canary_word_leaked(
+                user_input, response_completion, canary_word, log_outcome
+            )
+            assert not leak_detected
+
+
+def test_detect_injection_heuristics(
+    rebuff: RebuffSdk,
+    prompt_injection_inputs: List[str],
+    benign_inputs: List[str],
+    detect_injection_arguments: Dict,
+):
+    detect_injection_arguments["check_heuristic"] = True
+
+    for prompt_injection in prompt_injection_inputs:
+        rebuff_response = rebuff.detect_injection(
+            prompt_injection, **detect_injection_arguments
+        )
+        assert (
+            rebuff_response.heuristic_score
+            > detect_injection_arguments["max_heuristic_score"]
+        )
+        assert rebuff_response.injection_detected
+
+    for input in benign_inputs:
+        rebuff_response = rebuff.detect_injection(input, **detect_injection_arguments)
+        assert (
+            rebuff_response.heuristic_score
+            < detect_injection_arguments["max_heuristic_score"]
```

Comment:
Should this be `<=`? Not sure how likely it is for the heuristic_score to be equal to the max.
```suggestion
            <= detect_injection_arguments["max_heuristic_score"]
```

Reply:
Thank you, I have updated the comparisons as you suggested.